zebra-network/src/peer_set/candidate_set/tests/vectors.rs

```diff
@@ -135,3 +221,32 @@ fn mock_gossiped_peers(last_seen_times: impl IntoIterator<Item = DateTime<Utc>>)
         })
         .collect()
 }
+
+/// Create a mock `PeerSet` service that checks that requests to it are rate limited.
+fn mock_peer_service<E>(
+) -> impl Service<Request, Response = Response, Future = future::Ready<Result<Response, E>>, Error = E>
+       + 'static {
+    let rate_limit_interval = MIN_PEER_GET_ADDR_INTERVAL;
+
+    let mut peer_request_tracker: VecDeque<_> =
+        iter::repeat(Instant::now()).take(GET_ADDR_FANOUT).collect();
+
+    tower::service_fn(move |request| {
+        match request {
+            Request::Peers => {
+                // Get time from queue that the request is authorized to be sent
+                let authorized_request_time = peer_request_tracker
+                    .pop_front()
+                    .expect("peer_request_tracker should always have GET_ADDR_FANOUT elements");
+                // Check that the request was rate limited
+                assert!(Instant::now() >= authorized_request_time);
+                // Push a new authorization, updated by the rate limit interval
+                peer_request_tracker.push_back(authorized_request_time + rate_limit_interval);
```

Comment:
The new time should be based on `max(Instant::now(), authorized_request_time)`, not just the previous time.

The previous time could be a long way in the past, but the current time could also be paused during tests. So we need to use both.